File: api/src/app/useCases/product.ts
import { Request, Response } from 'express';
import { Product } from '../models/Product';

export async function listProducts(req: Request, res: Response) {
  try {
    const products = await Product.find();
    res.json(products);
  } catch (error) {
    res.status(500).json({ message: error });
  }
}

export async function createProduct(req: Request, res: Response) {
  try {
    const imagePath = req.file?.filename;
    const { name, description, price, category, ingredients } = req.body;
    const product = await Product.create(
      {
        name,
        description,
        price: Number(price),
        category,
        ingredients: ingredients ? JSON.parse(ingredients) : [],
        imagePath,
      },
    );
    res.json(product);
  } catch (error) {
    res.status(500).json({ error: error });
  }
}

File: mobile/lib/src/models/product.dart
class Product {
  final String? _id = null;
  final int idCategory;
  final String name;
  final String description;
  final String imagePath;
  final double price;
  int quantity = 1;
  final List<Ingredient> ingredients;

  get id => _id;

  Product({
    required String id,
    required this.idCategory,
    required this.description,
    required this.imagePath,
    required this.price,
    required this.name,
    required this.ingredients,
  });
}

class Ingredient {
  final String? _id = null;
  final String name;
  final String icon;

  get id => _id;

  Ingredient({
    required String id,
    required this.name,
    required this.icon,
  });
}
